arroyo-worker/src/connectors/kafka/sink/mod.rs

```diff
@@ -150,4 +235,40 @@ impl<K: Key + Serialize, T: Data + Serialize> KafkaSinkFunc<K, T> {
 
         self.publish(k, v).await;
     }
+
+    async fn handle_commit(&mut self, epoch: u32, ctx: &mut crate::engine::Context<(), ()>) {
+        let ConsistencyMode::ExactlyOnce { next_transaction_index: _, producer_to_complete } = &mut self.consistency_mode else {
+            warn!("received commit but consistency mode is not exactly once");
+            return
+        };
+
+        let Some(committing_producer)= producer_to_complete.take() else {
+            unimplemented!("received a commit message without a producer ready to commit. Restoring from commit phase not yet implemented");
+        };
+        committing_producer
+            .commit_transaction(Timeout::Never)
+            .expect("committing producer should succeed");
```

Comment:
does it make sense to have any retry logic here, given that a single failure is unrecoverable?

Reply:
I'm not sure I can think of something we could do to recover from an error here, without a bunch of the other retry management work we've discussed.